subworkflows/local/prok_mags_generation.nf

```diff
@@ -0,0 +1,105 @@
+/*
+    ~~~~~~~~~~~~~~~~~~~~~~~~
+     Prokaryotes subworkflow
+    ~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+include { REFINEMENT as BIN_REFINEMENT       } from './mgbinrefinder/binrefinder'
+include { CLEAN_AND_FILTER_BINS              } from './clean_and_filter_bins'
+
+include { CHECKM2                            } from '../../modules/local/checkm2/main'
+include { DREP                               } from '../../modules/local/drep/main'
+include { COVERAGE_RECYCLER                  } from '../../modules/local/coverage_recycler/main'
+include { DETECT_RRNA                        } from '../../modules/local/detect_rrna/main'
+include { GTDBTK                             } from '../../modules/local/gtdbtk/main'
+include { CHANGE_UNDERSCORE_TO_DOT           } from '../../modules/local/utils'
+
+
+process CHECKM_TABLE_FOR_DREP_GENOMES {
+
+    input:
+    path(checkm_filtered_genomes_dir)
+    path(dereplicated_genomes_tsv)
+
+    output:
+    path("checkm_results_mags.tab"), emit: checkm_results_mags
+
+    script:
+    """
+    grep -f ${dereplicated_genomes_tsv} ${checkm_filtered_genomes_dir} > checkm_results_mags.tab
+    """
+}
+
+
+workflow PROK_MAGS_GENERATION {
+
+    take:
+    collected_binners_and_depth // tuple( meta, concoct, metabat, maxbin, depth_file)
+    cat_db_folder
+    cat_diamond_db
+    cat_taxonomy_db
+    gunc_db
+    checkm2_db
+    gtdbtk_db
+    rfam_rrna_models
+
+    main:
+
+    collected_binners = collected_binners_and_depth.map { meta, concot_bins, maxbin_bins, metabat_bins, _ -> 
+        [ meta, concot_bins, maxbin_bins, metabat_bins ]
+    }
+
+    metabat_depth = collected_binners_and_depth.map { it -> it[4] }
+
+    // -- bin refinement //
+    BIN_REFINEMENT( collected_binners, checkm2_db )
+
+    // -- clean bins
+    CLEAN_AND_FILTER_BINS( 
+        BIN_REFINEMENT.out.bin_ref_bins,
+        cat_db_folder,
```

Comment:
what a magic, why no .first() 😮

Reply:
yeah, for parameters that are folders or files we shouldn't be using `channel.from`, we should use `file`. We then can remove the `.first()`